ci-operator/step-registry/confidential-clusters/ref/cocl/beaker-kind-provision/confidential-clusters-ref-cocl-beaker-kind-provision-commands.sh

```diff
@@ -0,0 +1,1283 @@
+# ============================================================================
+echo "[INFO] Patching scripts/common.sh to fix DOCKER_HOST..."
+echo "[INFO] Current directory: $(pwd)"
+echo "[INFO] Searching for scripts/common.sh..."
+find . -name "common.sh" -type f 2>/dev/null | head -5 || echo "  No common.sh found"
+
+if [ -f "scripts/common.sh" ]; then
+  echo "[INFO] Found scripts/common.sh, backing up and rewriting..."
+
+  # Backup original
+  cp scripts/common.sh scripts/common.sh.orig
+
+  echo "=========================================="
+  echo "[DEBUG] Original scripts/common.sh content:"
+  cat -n scripts/common.sh
+  echo "=========================================="
+
+  # Don't try to patch with sed - just completely rewrite it
+  # This avoids bash syntax errors from empty if blocks
+  echo "[INFO] Completely rewriting scripts/common.sh for root docker..."
+
+  cat > scripts/common.sh << 'COMMON_SH_FIXED'
+#!/bin/bash
+# Patched by CI to fix DOCKER_HOST issue
+# Root docker uses /var/run/docker.sock by default, no need to set DOCKER_HOST
+
+RUNTIME=${RUNTIME:-docker}
+
+# For podman, we need to set some environment variables
+if [ "$RUNTIME" == "podman" ]; then
+  export KIND_EXPERIMENTAL_PROVIDER=podman
+  if [ -n "${XDG_RUNTIME_DIR:-}" ]; then
+    export DOCKER_HOST=unix://${XDG_RUNTIME_DIR}/podman/podman.sock
+  fi
+fi
+
+# For docker, use defaults (no DOCKER_HOST needed for root docker)
+# Docker daemon socket is at /var/run/docker.sock by default
+COMMON_SH_FIXED
+
+  echo "=========================================="
+  echo "[DEBUG] New scripts/common.sh content:"
+  cat -n scripts/common.sh
+  echo "=========================================="
+
+  echo "[INFO] Diff between original and new:"
+  diff -u scripts/common.sh.orig scripts/common.sh || true
+
+  echo "[SUCCESS] scripts/common.sh rewritten successfully"
+else
+  echo "[WARN] scripts/common.sh not found in ${WORK_DIR}"
+  echo "[INFO] Creating a minimal scripts/common.sh for docker..."
+  mkdir -p scripts
+  cat > scripts/common.sh << 'COMMON_SH_NEW'
+#!/bin/bash
+# Created by CI - minimal common.sh for docker
+RUNTIME=${RUNTIME:-docker}
+echo "[INFO] Runtime set to: ${RUNTIME}"
+# For root docker, no special environment variables needed
+COMMON_SH_NEW
+  chmod +x scripts/common.sh
+  echo "[SUCCESS] Created minimal scripts/common.sh"
+fi
+
+# Verify the file is syntactically correct
+echo "[INFO] Verifying scripts/common.sh syntax..."
+if bash -n scripts/common.sh; then
+  echo "[SUCCESS] scripts/common.sh has valid bash syntax"
+else
+  echo "[ERROR] scripts/common.sh has syntax errors!"
+  cat -n scripts/common.sh
+  exit 1
+fi
+
+echo "[SUCCESS] scripts/common.sh is ready for docker"
+
+# ============================================================================
+# Patch all scripts to fix podman-specific flags for docker
+# ============================================================================
```

Comment:
Please don't patch the script in this way, it will make the debugging very hard if you modify what we have in the main repository on the flight. If you need some extra flag, please add the changes in the original repository